NeosModLoader/AssemblyHider.cs

```diff
@@ -64,20 +70,23 @@ private static HashSet<Assembly> GetModAssemblies()
 			return assemblies;
 		}
 
-		private static bool IsModType(Type type)
+		private static bool IsModAssembly(Assembly assembly, string typeOrAssembly, string name, bool log, bool forceShowLate)
 		{
-			if (neosAssemblies!.Contains(type.Assembly))
+			if (neosAssemblies!.Contains(assembly))
 			{
 				// the type belongs to a Neos assembly
-				return false; // don't hide the type
+				return false; // don't hide the thing
 			}
 			else
 			{
-				if (modAssemblies!.Contains(type.Assembly))
+				if (modAssemblies!.Contains(assembly))
 				{
 					// known type from a mod assembly
-					Logger.DebugInternal($"Hid type \"{type}\" from Neos");
-					return true; // hide the type
+					if (log)
```

Comment:
Logger already has `DebugFuncInternal` which does a condition check, does this really need it's own separate logging control..?

Reply:
I've got a comment further down where I invoke with log=false `// this generates a lot of logspam, as a single call to AppDomain.GetAssemblies() calls this many times`

So essentially if you have debug logging turned on then this will emit a debug log line for every loaded assembly each time Froox calls `AppDomain.GetAssemblies()`... which he does *a lot*.  My thoughts are that the `Logger.DebugFuncInternal(() => $"Intercepting call to AppDomain.GetAssemblies() from {callingAssembly}");` further down is all the debug logging we really need in this case, so I added a mechanism to turn off the per-assembly debug logs.